airflow/ui/src/pages/DagsList/Dag/Code/Code.tsx

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Button, Heading, HStack } from "@chakra-ui/react";
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { PrismLight as SyntaxHighlighter } from "react-syntax-highlighter";
+import python from "react-syntax-highlighter/dist/esm/languages/prism/python";
+import {
+  oneLight,
+  oneDark,
+} from "react-syntax-highlighter/dist/esm/styles/prism";
+
+import {
+  useDagServiceGetDagDetails,
+  useDagSourceServiceGetDagSource,
+} from "openapi/queries";
+import { ErrorAlert } from "src/components/ErrorAlert";
+import Time from "src/components/Time";
+import { ProgressBar } from "src/components/ui";
+import { useColorMode } from "src/context/colorMode";
+
+SyntaxHighlighter.registerLanguage("python", python);
+
+export const Code = () => {
+  const { dagId } = useParams();
+
+  const {
+    data: dag,
+    error,
+    isLoading,
+  } = useDagServiceGetDagDetails({
+    dagId: dagId ?? "",
+  });
+
+  const {
+    data: code,
+    error: codeError,
+    isLoading: isCodeLoading,
+  } = useDagSourceServiceGetDagSource(
+    {
+      fileToken: dag?.file_token ?? "",
+    },
+    undefined,
+    { enabled: Boolean(dag?.file_token) },
+  );
+
+  // TODO: get default_wrap from config
+  const [wrap, setWrap] = useState(false);
+
+  const toggleWrap = () => setWrap(!wrap);
+  const { colorMode } = useColorMode();
+
+  const style = colorMode === "dark" ? oneDark : oneLight;
+
+  // wrapLongLines wasn't working with the prsim styles so we have to manually apply the style
+  if (style['code[class*="language-"]'] !== undefined) {
```

Comment:
This looks like it doesn't fix the issue as I tried applying wrap but not sure if it's different to the fix applied here. There is a PR open in the repo but not merged.

Issue : https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/376
PR : https://github.com/react-syntax-highlighter/react-syntax-highlighter/pull/483

Sample dag : 

```python
import pendulum
from datetime import datetime, timedelta, timezone

from airflow import DAG
from airflow.decorators import task

# This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment

with DAG(
    dag_id="wrap_dag",
    start_date=datetime(2024, 1, 1),
    end_date=datetime(2024, 12, 12),
    schedule=timedelta(minutes=1),
) as dag:

    @task
    def this_is_a_very_long_function_name_to_enable_wrapping(ds=None):
        """
        This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment
        """
        pass

    @task
    def this_is_a_very_long_function_name_to_enable_wrapping_1(ds=None):
        """
        This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment
        """
        pass

    @task
    def this_is_a_very_long_function_name_to_enable_wrapping_2(ds=None):
        """
        This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment
        """
        pass


    this_is_a_very_long_function_name_to_enable_wrapping() >> this_is_a_very_long_function_name_to_enable_wrapping_1() >> this_is_a_very_long_function_name_to_enable_wrapping_2()
```

![image](https://github.com/user-attachments/assets/2529a0c0-a285-4d64-8bfd-f10ee1e77505)

Reply:
Yes, we needed to apply two fixes. I just added the custom renderer in that issue thread. Now its working correctly:

<img width="1579" alt="Screenshot 2024-11-05 at 11 47 12 AM" src="https://github.com/user-attachments/assets/ae470d53-75ed-4b67-ba45-0f3006a4e9b4">
<img width="1584" alt="Screenshot 2024-11-05 at 11 47 20 AM" src="https://github.com/user-attachments/assets/44a2dd59-5fc9-4298-bdea-4d4d6dc437b2">

Reply:
Thanks @bbovenzi, tested the latest commit and can confirm it's working fine.